File: KalimaJavaExample/src/org/kalima/example/client/KalimaServerCallBack.java
package org.kalima.example.client;

import java.nio.channels.SocketChannel;

import org.kalima.kalimamq.message.KMessage;
import org.kalima.kalimamq.nodelib.Node;
import org.kalima.kalimamq.nodelib.ServerCallback;

public class KalimaServerCallBack implements ServerCallback {

	public KalimaServerCallBack() {	}

	@Override
	public void putData(SocketChannel socket, KMessage kMessage) {}

	@Override
	public void processSnapshot(SocketChannel ch, KMessage kMessage) {}

	@Override
	public void disconnected(String name, int port) {}

	@Override
	public void nodeAcked(String nodeName, String hostName, int port) {}

	@Override
	public void nodeDisconnected(Node node, boolean b) {}

	@Override
	public void nodeConnected(Node node) {}

	@Override
	public void onConnectionChanged(int status) {}

	@Override
	public void addCache(String cachePath) {}

	@Override
	public void removeCache(String cachePath) {}

	@Override
	public void processSnapshot(SocketChannel soc, String cachePathSeqs) {}

	@Override
	public KMessage processNextMessage(KMessage msg, SocketChannel soc) {
		return null;
	}

	@Override
	public void onVoteDone(boolean isLeader) {
		// TODO Auto-generated method stub
		
	}
}

File: KalimaCSharpExample/KalimaCSharpExample/KalimaServerCallBack.cs
using System;
using org.kalima.kalimamq.nodelib;
using org.kalima.kalimamq.message;
using java.nio.channels;

namespace KalimaCSharpExample
{
	public class KalimaServerCallBack : ServerCallback
	{
		public KalimaServerCallBack (){}

		public void putData(SocketChannel socket, KMessage kMessage) {}
			
		public void processSnapshot(SocketChannel ch, KMessage kMessage) {}

		public void disconnected(String name, int port) {}
	
		public void nodeAcked(String nodeName, String hostName, int port) {}

		public void nodeDisconnected(Node node, bool b) {}
	
		public void nodeConnected(Node node) {}

		public void onConnectionChanged(int status) {}

		public void addCache(String cachePath) {}
			
		public void removeCache(String cachePath) {}

		public void processSnapshot(SocketChannel soc, String cachePathSeqs, bool fromNotaryNode) {}
			
		public KMessage processNextMessage(KMessage msg) {
			return null;
		}
			
		public void onVoteDone(bool isLeader) {}
	}
}

